lib/ui/painting/image_decoder_no_gl_unittests.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/image_decoder_no_gl_unittests.h"
+
+namespace flutter {
+namespace testing {
+
+// Tests are disabled for fuchsia.
+#if defined(OS_FUCHSIA)
+#pragma GCC diagnostic ignored "-Wunreachable-code"
+#endif
+
+namespace {
+
+bool IsPngWithPLTE(const uint8_t* bytes, size_t size) {
+  if (size < 8) {
+    return false;
+  }
+
+  if (memcmp(bytes, "\x89PNG\x0d\x0a\x1a\x0a", 8) != 0) {
+    return false;
+  }
+
+  const uint8_t* end = bytes + size;
+  const uint8_t* loc = bytes + 8;
+  while (loc + 8 <= end) {
+    uint32_t chunk_length =
+        (loc[0] << 24) | (loc[1] << 16) | (loc[2] << 8) | loc[3];
+
+    if (memcmp(loc + 4, "PLTE", 4) == 0) {
+      return true;
+    }
+
+    loc += /*length*/ 4 + /*type*/ 4 + chunk_length + /*crc*/ 4;
+  }
+
+  return false;
+}
+
+sk_sp<SkData> OpenFixtureAsSkData(const char* name) {
```

Comment:
There are several copies of `OpenFixtureAsSkData` in various tests.

Can they be consolidated into someplace like https://github.com/flutter/engine/blob/main/testing/testing.cc?

(I'd be fine with doing this in a separate PR if it's nontrivial)

Reply:
I didn't mean to add one more, removed the duplication I introduced.